### PR TITLE
refactor: replace python-jose with pyjwt

### PR DESCRIPTION
**Pull request by mariuswinger**
## Why is this pull request needed?
The package python-jose seems to be unmaintained (last commit in 2021). This PR replaces it with a more up to date alternative. Courtesy of @soofstad

## What does this pull request change?
Just replaces the relevant parts of code. Replacement was fairly straight-forward.

### CODE REVIEW

File: api/src/authentication/authentication.py [hunk 1]
```diff
@@ -18,18 +18,12 @@
 
 
 @cached(cache=TTLCache(maxsize=32, ttl=86400))
-def fetch_openid_configuration() -> dict[str, str]:
+def fetch_openid_configuration() -> jwt.PyJWKClient:
```
Comment (soofstad) [inline api/src/authentication/authentication.py:21]:
```suggestion
def get_JWK_client() -> jwt.PyJWKClient:
```

**Review by soofstad — COMMENTED** [review]
I have only one question, does the HTTP request to grab JWKs get cached?
Like, does that happen on JWKClient instantiation, or when we first access the keys.

**Comment by mariuswinger** [comment]
> I have only one question, does the HTTP request to grab JWKs get cached? Like, does that happen on JWKClient instantiation, or when we first access the keys.

From what I can gather caching to the client happens when a signing key is requested. But we do create a new client for every request, so that would likely not make use of that caching ...

**Comment by mariuswinger** [comment]
> > I have only one question, does the HTTP request to grab JWKs get cached? Like, does that happen on JWKClient instantiation, or when we first access the keys.
> 
> From what I can gather caching to the client happens when a signing key is requested. But we do create a new client for every request, so that would likely not make use of that caching ...

Nope, there is a cache decorator above the get_JWK_client function, so it returns the same client instance and therefore utilizes the caching!

**Review by soofstad — APPROVED** [review]

**Pull request merged**